maker/src/orderbook_ws.rs

```diff
@@ -39,16 +39,17 @@ pub struct Client {
 
 impl Client {
     pub fn new(
-        endpoint: Url,
+        mut endpoint: Url,
         trader_id: PublicKey,
         auth_sk: SecretKey,
         position_manager: xtra::Address<position::Manager>,
         orderbook_status: watch::Sender<ServiceStatus>,
     ) -> Self {
-        let domain = endpoint.domain().expect("domain");
-        let port = endpoint.port().expect("port");
-
-        let url = format!("ws://{domain}:{port}/api/orderbook/websocket");
+        endpoint
+            .set_scheme("ws")
```

Comment:
We will need to distinguish between `ws` and `wss` at some point here. I guess this logic is assuming that we are using http.